clients/venice-admin-tool/src/main/java/com/linkedin/venice/AdminTool.java

```diff
@@ -1803,14 +1804,33 @@ private static void deleteKafkaTopic(CommandLine cmd, PubSubClientsFactory pubSu
 
   private static void dumpAdminMessages(CommandLine cmd, PubSubClientsFactory pubSubClientsFactory) {
     Properties consumerProperties = loadProperties(cmd, Arg.KAFKA_CONSUMER_CONFIG_FILE);
+    VeniceProperties veniceProperties = new VeniceProperties(consumerProperties);
     String pubSubBrokerUrl = getRequiredArgument(cmd, Arg.KAFKA_BOOTSTRAP_SERVERS);
     consumerProperties = DumpAdminMessages.getPubSubConsumerProperties(pubSubBrokerUrl, consumerProperties);
+    PubSubPositionTypeRegistry pubSubPositionTypeRegistry =
+        PubSubPositionTypeRegistry.fromPropertiesOrDefault(veniceProperties);
+    PubSubPositionDeserializer pubSubPositionDeserializer = new PubSubPositionDeserializer(pubSubPositionTypeRegistry);
     PubSubConsumerAdapter consumer = getConsumer(consumerProperties, pubSubClientsFactory);
+    String startingOffset = getOptionalArgument(cmd, Arg.STARTING_OFFSET);
```

Comment:
We should do argument validation before constructing other main dependencies like consumer.